src/main/java/com/example/codeE/controller/UserController.java

```diff
@@ -17,9 +22,25 @@ public class UserController {
 
     @GetMapping
     @RequestMapping(value = "",method = RequestMethod.GET)
-    public ResponseEntity<?> getAllUsers(){
-        List<User> listaPersona = this.userImplement.getAllUsers();
-        return ResponseEntity.ok(listaPersona);
+    public ResponseEntity<?> getAllUsers(
+            @RequestParam(defaultValue = "") String role,
+            @RequestParam(defaultValue = "") String searchKeyword,
+            @RequestParam(defaultValue = "0") Integer pageNumber,
+            @RequestParam(defaultValue = "10") Integer pageSize
```

Comment:
@Noro911, can we do some validation here?

Please check this: https://www.baeldung.com/spring-validate-requestparam-pathvariable

The default value of pageNumber should be 1 instead of 0.